chrome-extension/content.js

```diff
@@ -189,14 +198,59 @@ async function getCategoryDropdown(userId) {
   return categoryDropdown;
 }
 
+/**
+ * Returns select HTML element of places search result options
+ * @param {String} text 
+ */
+async function getPlacesSelection(text) {
+  const searchResults = await placesSearch(text);
+  if (searchResults.length > 0) {
+    const placesSelection = customCreateElement('select', []);
+    
+    const defaultOption = customCreateElement('option', [], 'Optional: Select a Known Location');
+    defaultOption.value = '';
+    defaultOption.selected = true;
+    defaultOption.disabled = true;
+    placesSelection.options.add(defaultOption);
+
+    searchResults.map((place) => {
+      const newOption = customCreateElement('option', [], place.name);
+      newOption.value = place.place_id
+      placesSelection.options.add(newOption);
+    })
+    return placesSelection;
+  }
+  return null;
+}
+
+/**
+ * Helper function that sends a request to Google Places Search and returns a list of
+ * search results, if there are.
+ * @param {String} text 
+ */
+async function placesSearch(text) {
+  try {
+    const response = await fetch(
+      serverUrl + '/extension/placesSearch/' + text
+    );
+    const candidates = await response.json();
+    console.log('response', candidates);
+    return candidates
+  } catch (error) {
+    console.log('heres an error from places search');
+    console.log(error);
+    return null;  
+  }
+}
+
 /**
  * Add new item to database
  * Side bar turns to success message panel once item successfully added  
  */
 async function addItem(mongoId) {
   const newItem = {
     sourceLink: window.location.toString(),
-    placesId: 'something', // TODO: get actual placesId
+    placesId: document.getElementById('placesSelector').value,
```

Comment:
In your function above, placesSelection might not be added to the page above so then you couldn't access the value field of a missing element. So you should probably add a check for whether the field exists.